source/ncengine/serialize/ComponentSerialization.cpp

```diff
@@ -179,9 +179,24 @@ void SerializeRigidBody(std::ostream& stream, const RigidBody& out, const Serial
     serialize::Serialize(stream, ctx.entityMap.at(out.GetEntity()));
 
     const auto& shape = out.GetShape();
-    serialize::Serialize(stream, shape.GetType());
-    serialize::Serialize(stream, shape.GetLocalScale());
-    serialize::Serialize(stream, shape.GetLocalPosition());
+    const auto shapeType = shape.GetType();
+    serialize::Serialize(stream, shapeType);
+    switch (shapeType)
+    {
+        case ShapeType::Box:
+        case ShapeType::Sphere:
+        case ShapeType::Capsule:
+            serialize::Serialize(stream, shape.GetLocalScale());
+            serialize::Serialize(stream, shape.GetLocalPosition());
+            break;
+        case ShapeType::ConvexHull:
+            serialize::Serialize(stream, shape.GetAssetId());
+            serialize::Serialize(stream, shape.GetLocalScale());
```

Comment:
Why don't we care about position for hulls?

Reply:
If we really need it, it can be added in the future. I didn't do it now because translation/rotation/scaling of shapes in jolt is done via various `DecoratedShape` subclasses (you construct a base instance of the actual shape, from that you make a scaled shape instance, ...). For primitives, we can use scale directly in the construction of the base instance, resulting in only one 'wrapper' shape for translation. For hull/mesh shapes, we cannot, so we get sort of a nesting doll situation. Also, I figured that the main use of hull/mesh shapes is to match/approximate a rendered mesh. In those cases, position is handled by the local transform values.